cmake/CeleritasUtils.cmake

```diff
@@ -266,45 +304,56 @@ endfunction()
 #-----------------------------------------------------------------------------#
 
 function(celeritas_add_library target)
-  celeritas_rdc_add_library(${target} ${ARGN})
+  
+  cuda_get_sources_and_options(_sources _cmake_options _options ${ARGN})
+  cuda_rdc_sources_contains_cuda(_cuda_sources ${_sources})
+  if(CELERITAS_USE_HIP AND _cuda_sources)
+    # When building Celeritas libraries, we put HIP/CUDA files in shared .cu
+    # suffixed files. Override the language if using HIP.
+    set_source_files_properties(
+      ${_cuda_sources}
+      PROPERTIES LANGUAGE HIP
+    )
+  endif()
+# Potential speed-up (if it is a significant gain should it be applied to the
+# other calls to?)
+#  if(NOT CELERITAS_USE_VecGeom OR NOT CMAKE_CUDA_COMPILER OR NOT _cuda_sources)
+#    add_library(${target} ${ARGN})
+# else()
```

Comment:
Yes, I think we should fall through in non-vecgeom cases, here and the other macros, because it will produce fewer edge cases if we're trying to build on some obscure system or with an unexpected cmake change, etc.

Reply:
One option for this is 007ac62c4c0473c95ecc0ae256d9c3b97f41e374 or we can add more 5 liners wrappers ... neither is very attractive.

Reply:
I think 5 liners is fine.